File: Daybreak.GWCA/source/payloads/StatePayload.cpp
#include <cstdint>
#include <json.hpp>
#include <payloads/StatePayload.h>

using json = nlohmann::json;

namespace Daybreak {
    void to_json(json& j, const StatePayload& p) {
        j = json
        {
            {"Id", p.Id},
            {"PosX", p.PosX},
            {"PosY", p.PosY},
            {"State", p.State},
            {"Health", p.Health},
            {"Energy", p.Energy}
        };
    }
}

File: Daybreak/Services/Scanner/Models/StatePayload.cs
namespace Daybreak.Services.Scanner.Models;
internal sealed class StatePayload
{
    public uint Id { get; set; }
    public float PosX { get; set; }
    public float PosY { get; set; }
    public uint State { get; set; }
    public float Health { get; set; }
    public float Energy { get; set; }
}
